### DISCUSSION TITLE
set a value of porepressure of a particular block?

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#18090] [Q&A Modules: General] [answered]
Hi guys,

is it possible to set a value of pore-pressure of a particular block (sub-domain), maybe via [Materials] block?

I know I can define the boundaries of a sub-domain and then set the boundary of that block to zero (below).

However, when I have like 100+ sub-domains, my input file gets real ugly.

Note: I am currently using `PorousFlowFullySaturated` action

Kind regards,
Traiwit

```
  [add_side_sets_block6]
    type = SideSetsAroundSubdomainGenerator
    block = 6
    input = add_side_sets
    fixed_normal = true
    new_boundary = 'front6 top6 right6 back6 bottom6 left6'
    variance = 0.5
  []

  [./water_grad_block6]
     type = DirichletBC
     variable = porepressure
     boundary = 'front6 top6 right6 back6 bottom6 left6'
     value = 0
  [../]
```

**Comment by WilkAndy** [comment, marked as answer]
Yep, usually input files are looooong when you have lots of subdomains because lots of things are going on.   i usually write a python script to write my input files, to avoid mistakes.

**Reply by Traiwit** [reply]
Hi @WilkAndy, this is probably the easiest way to deal with the input file. Thanks!

**Comment by permcody** [comment]
The other solution to this problem is to create a custom "Action". Actions allow you to programmatically setup your problem (add variables, kernels, bcs, other objects). You can then replace a long list of complicated blocks with a much smaller, higher-level input block for your application.

**Comment by GiudGiud** [comment]
For this example, if it's 0 DirichletBC all the time for all pores, then you could:
- add many blocks to SideSetsAroundSubdomainGenerator to have all the sidesets (can one sideset cover them all? I m not sure)
- add all the boundaries (or just one if that worked) to the DirichletBC

**Reply by Traiwit** [reply]
@GiudGiud thank you for your reply, I was thinking about that too, but it wouldn't work if I want to change the properties of each block at each particular time.
